### ISSUE TITLE
Version 001 Refactor for Typescript & Microbundle

### ISSUE THREAD

**Issue by pBread** [mistralai/client-js#40]
Hi All,

I was playing around with the JS client and ran into a few environment compatibility issues, e.g. a conflict to do w/JS’s lovely module declaration rules.

I started writing a fix and, well, I ended up refactoring most it. It should be completely backward compatible, i.e. the methods are named the same and behave the same. 

I’ve managed other NPM libraries before, all for employers, and I believe this is a more solid foundation than what had been originally undertaken. It's in the [v001-rewrite-ts-microbundle](https://github.com/pBread/client-js/tree/v001-rewrite-ts-microbundle) branch of my fork [pbread/client-js](https://github.com/pBread/client-js).

_Note: As I write this, I have the client rewritten but I still need to port over the tests and examples. I'll be doing that now but there might be some minor bugs I need to work through._

Here are the major enhancements:
**Native Typescript Support** It’s quite tedious to make a native JS compatible with TS. Going from TS to JS is easier and you benefit from type-safety.

**Microbundle:** [Microbundle](https://github.com/developit/microbundle) is wonderful. There almost zero configuration and it takes care of polyfill, minification, cross environment compatibility, TS declaration generation, etc. etc. 

Microbundle solves the original problem I ran into. And, it will immediately add functionality, such as producing the UMD files necessary for browser <script> support.

**Replaced node-fetch w/isomorphic-fetch:** [isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch) is just a wrapper around node-fetch that solves the globalThis.fetch compatibility problem.

**Implemented private keyword:** TS offers typesafety but it doesn't actually protect the property. This means the apiKey property can be sniffed easily by any other code in the environment. We can easily add some protection here. I'll make another issue for that.

**Added Versioned Facade:** Mistral's API is obviously very new. I expect the routes and parameters will change drastically in the coming months. I've added a versioned facade in front of the root-level methods which will allow the client to add temporary fixes to breaking changes, version itself, etc. 

Basically, the class has a private v1 property that aligns with Mistral's route structure then the root-level methods call those methods.

`client.listModels === client.v1.models.list`

**Comment by pBread** [comment]
I just submitted a pull request with the changes. I can't link this issue to that request so I'm going to delete close this issue and take up the discussion there.
